cranelift/isle/isle/src/lexer.rs

```diff
@@ -247,6 +244,10 @@ impl<'src> Lexer<'src> {
     pub fn eof(&self) -> bool {
         self.lookahead.is_none()
     }
+
+    fn peek_byte(&self) -> Option<u8> {
+        self.src.as_bytes().get(self.pos.offset).copied()
+    }
```

Comment:
Why not `fn peek_char(&self) -> Option<char>`? I guess a byte is mildly more performant, in theory if LLVM doesn't clean things up, but that doesn't seem super compelling compared to the simplification of "we are dealing with strings, so we should also deal with chars".